src/network/networkd-mptcp.c

```diff
@@ -0,0 +1,400 @@
+                link->mp_tcp_configured = true;
+                link_check_ready(link);
+        }
+
+        return 1;
+}
+
+static int mp_tcp_configure_address(Link *link, MPTCP *mp_tcp) {
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
+        int r;
+
+        assert(link);
+        assert(link->manager);
+        assert(link->manager->rtnl);
+        assert(link->ifindex > 0);
+
+        r = sd_genl_message_new(link->manager->genl, SD_GENL_MPTCP, MPTCP_PM_CMD_ADD_ADDR, &req);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Failed to allocate generic netlink message: %m");
+
+        r = sd_netlink_message_open_container(req, MPTCP_PM_ATTR_ADDR);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not open MPTCP_PM_ATTR_ADDR container: %m");
+
+        if (mp_tcp->id_is_set) {
+                r = sd_netlink_message_append_u8(req, MPTCP_PM_ADDR_ATTR_ID, mp_tcp->id);
+                if (r < 0)
+                        return log_link_error_errno(link, r, "Could not append MPTCP_PM_ADDR_ATTR_ID attribute: %m");
+        }
+
+        r = sd_netlink_message_append_s32(req, MPTCP_PM_ADDR_ATTR_IF_IDX, link->ifindex);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not append MPTCP_PM_ADDR_ATTR_IF_IDX attribute: %m");
+
+        if (in_addr_is_null(mp_tcp->family, &mp_tcp->address) == 0) {
+                r = sd_netlink_message_append_u16(req, MPTCP_PM_ADDR_ATTR_FAMILY, mp_tcp->family);
+                if (r < 0)
+                        return log_link_error_errno(link, r, "Could not append MPTCP_PM_ADDR_ATTR_FAMILY attribute: %m");
+
+                if (mp_tcp->family == AF_INET)
+                        r = sd_netlink_message_append_in_addr(req, MPTCP_PM_ADDR_ATTR_ADDR4, &mp_tcp->address.in);
+                else
+                        r = sd_netlink_message_append_in6_addr(req, MPTCP_PM_ADDR_ATTR_ADDR6, &mp_tcp->address.in6);
+                if (r < 0)
+                        return log_link_error_errno(link, r, "Could not append MPTCP_PM_ADDR_ATTR_ADDR4 / MPTCP_PM_ADDR_ATTR_ADDR6 attribute: %m");
+        }
+
+        r = sd_netlink_message_close_container(req);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not close MPTCP_PM_ATTR_ADDR container: %m");
+
+        r = netlink_call_async(link->manager->genl, NULL, req, mp_tcp_handler,
+                               link_netlink_destroy_callback, link);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not send rtnetlink message: %m");
+
+        link_ref(link);
+        link->mp_tcp_messages++;
+
+        return 0;
+}
+
+static int mp_tcp_limit_handler(sd_netlink *rtnl, sd_netlink_message *n, Manager *m) {
+        int r;
+
+        assert(n);
+        assert(m);
+
+        r = sd_netlink_message_get_errno(n);
+        if (r < 0 && r != -EINVAL) {
```

Comment:
Same here. When kernel returns `-EINVAL`?

Reply:
After this runs if we restart networkd rather than giving exist it gives `EINVAL`

Reply:
`mptcp_nl_cmd_set_limits()` returns `-EINVAL` only when a limit is larger than 8 (`MPTCP_PM_ADDR_MAX`). Such error should not be ignored. Please drop the condition `r != -EINVAL`.